### PR TITLE
tutorial : prebuilt docker images

### PR DESCRIPTION
**Pull request by shivanissambare**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 1]
```diff
@@ -0,0 +1,938 @@
...
+          "start_time": "2021-05-18T20:36:44.722969",
+          "status": "completed"
+        },
+        "tags": []
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Copyright (c) Microsoft Corporation. All rights reserved.\n",
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:40]: remove license cell

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 2]
```diff
@@ -0,0 +1,938 @@
...
+        "\n",
+        "Licensed under the MIT License."
+      ],
+      "metadata": {
+        "nteract": {
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:45]: should remove this metadata and other extraneous things like by Azure ML notebooks

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 3]
```diff
@@ -0,0 +1,938 @@
...
+        },
+        "tags": []
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Prerequisites\n",
+        "\n",
+        "If you are using an Azure Machine Learning Notebook VM, you are all set. Otherwise, go through the [Configuration](../../../../configuration.ipynb) notebook to install the Azure Machine Learning Python SDK and create an Azure ML `Workspace`"
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:89]: notebook VMs are deprecated

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 4]
```diff
@@ -0,0 +1,938 @@
...
+          "status": "completed"
+        },
+        "tags": []
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Check core SDK version number\n",
+        "import azureml.core\n",
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:111]: remove this cell

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 5]
```diff
@@ -0,0 +1,938 @@
...
+          "start_time": "2021-05-18T20:36:45.491041",
+          "status": "completed"
+        },
+        "tags": []
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Initialize workspace\n",
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:144]: simplify, get to the interesting part of the tutorial sooner (assume the user has read the python-sdk/README.md, has a workspace and compute targets)

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 6]
```diff
@@ -0,0 +1,938 @@
...
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from azureml.core.workspace import Workspace\n",
+        "\n",
+        "ws = Workspace.from_config()\n",
+        "print(\n",
+        "    \"Workspace name: \" + ws.name,\n",
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:170]: just print out the object, remove the big print statement

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 7]
```diff
@@ -0,0 +1,938 @@
...
+        },
+        "tags": []
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Register model\n",
+        "\n",
+        "Register a file or folder as a model by calling [Model.register()](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.model.model?view=azure-ml-py#register-workspace--model-path--model-name--tags-none--properties-none--description-none--datasets-none--model-framework-none--model-framework-version-none--child-paths-none--sample-input-dataset-none--sample-output-dataset-none--resource-configuration-none-)\n"
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:208]: our SDK docs are often incorrect or misleading, and our links often break - would recommend removing

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 8]
```diff
@@ -0,0 +1,938 @@
...
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Write the requirements.txt\n",
+        "Write the `requirements.txt`\n",
+        "\n",
+        "> **Pin each package version:**\n",
+        "Even while prototyping, pin each package version in requirements.txt. \n",
+        "For example, use scipy == 1.2.3 instead of just scipy or even scipy > 1.2.3. \n",
+        "If you don't pin an exact version and scipy releases a new version, this can break your scoring script and cause failures during deployment and scaling."
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:297]: while true for production, for our examples we leave azureml-* packages unpinned to continually test on the latest version and ensure no breaking changes to our public-facing examples

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 9]
```diff
@@ -0,0 +1,938 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Define the environment (Dynamic installation via requirements.txt)\n",
+        "\n",
+        "Then, we will need to create an Azure Machine Learning environment that specifies all of the scoring script's package dependencies. \n",
+        "Here we assume the `requirements.txt` is in the right format and inlcudes all additional packages for the model.\n",
+        "\n",
+        "If deployment fails because of Python extensibility solution, there will be no logs available the [Azure Machine Learning Studio](https://ml.azure.com),`service.get_logs()` will return None. If there is a problem in the *init()* function of score.py, `service.get_logs()` will return logs for the same.\n",
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:355]: lowercase "studio", "Studio" refers to the sunset Studio classic product

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb [hunk 10]
```diff
@@ -0,0 +1,938 @@
...
+        "tags": []
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from azureml.core import Environment\n",
+        "from azureml.core.conda_dependencies import CondaDependencies \n",
+        "\n",
+        "myenv = Environment(name='my_pytroch_env')\n",
```
Comment (lostmygithubaccount) [inline python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:386]: see naming guidance, preferably something like "pytorch-deploy-prebuilt-example"

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]
various issues, generally reduce the number of cells and unnecessary or likely to break documentation/prose - in-depth explanations can go to docs.microsoft.com

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/requirements.txt [hunk 11]
```diff
@@ -0,0 +1,4 @@
+  torchvision==0.7.0
+  future==0.17.1
+  pillow==8.1.1
+  azureml.core==1.25.0
```
[the remark this answers is not in the document]
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/requirements.txt:4]: it goes against the guidance we are giving to our customers
not pinning a version may cause scoring script to break
I am re-submitting the PR with latest azureml core version

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 1 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:40]: done

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 2 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:45]: done

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 3 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:89]: oh wow, I was taking inspiration from these notebooks and they still refer it as notebook vm
we should definitely update these notebooks as well

example:

https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/ml-frameworks/tensorflow/train-hyperparameter-tune-deploy-with-tensorflow/train-hyperparameter-tune-deploy-with-tensorflow.ipynb

https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/ml-frameworks/pytorch/train-hyperparameter-tune-deploy-with-pytorch/train-hyperparameter-tune-deploy-with-pytorch.ipynb

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 6 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:170]: done

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 7 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:208]: done

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 8 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:297]: i am submitting pr with latest version
not pinning the version goes against the guidance

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 9 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:355]: done

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb (hunk 10 above)
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/pytorch-prebuilt-docker-image.ipynb:386]: changed

File: python-sdk/tutorials/deploy-with-prebuilt-docker-images/README.md [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# Deploy a model using Prebuilt docker image for Inference
```
[the remark this answers is not in the document]
Comment (shivanissambare) [reply python-sdk/tutorials/deploy-with-prebuilt-docker-images/README.md:1]: adding

**Comment by shivanissambare** [comment]
@lostmygithubaccount  Cody, please take a look tried to incorporated all the feedback

**Comment by ChrisMissal** [comment]
This appears to have broken the link here: https://docs.microsoft.com/en-us/azure/machine-learning/concept-prebuilt-docker-images-inference

**Pull request merged**